src/types.rs

```diff
@@ -170,6 +238,17 @@ impl From<BlsPublicKey> for PublicKeyOrIndex {
     }
 }
 
+impl From<String> for PublicKeyOrIndex {
```

Comment:
can implement `TryFrom` here instead and be able to naturally integrate an error type